dashboard/assets/components/Dashboard.jsx

```diff
@@ -0,0 +1,146 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import {withStyles} from 'material-ui/styles';
+
+import SideBar from './SideBar.jsx';
+import Header from './Header.jsx';
+import Main from "./Main.jsx";
+import {isNullOrUndefined, defaultZero, MEMORY_SAMPLE_LIMIT, TAGS} from "./Common.jsx";
+
+// Styles for the Dashboard component.
+const styles = theme => ({
+    appFrame: {
+        position: 'relative',
+        display: 'flex',
+        width: '100%',
+        height: '100%',
+        background: '#303030',
+    },
+});
+
+// Dashboard is the main component, which renders the whole page,
+// makes connection with the server and listens for messages.
+// When there is an incoming message, updates the page's content correspondingly.
+class Dashboard extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            active: TAGS.home.id, // active menu
+            sideBar: true, // true if the sidebar is opened
+            memory: [],
+            traffic: [],
+            logs: [],
+        };
```

Comment:
If possible, please align struct initializers. An automated tool is even better, a CI linter is possibly the best.

Reply:
This rule will handle it
https://eslint.org/docs/rules/key-spacing